HMRC-1438: Update FAQ Feedback Routing

amberstarlight opened the pull request:
### Jira link

[HMRC-1438](https://transformuk.atlassian.net/browse/HMRC-1438)

### What?

- Updated the Green Lanes FAQ feedback to use RESTful routing
- Updated feedback paths to align with https://github.com/trade-tariff/trade-tariff-backend/pull/2608

### Why?

These changes are necessary to align the frontend with the changed behaviour of the backend.

### Deployment risks

- We must create and apply a new environment variable, `BACKEND_API_TOKEN`, which provides a shared auth between the frontend and backend separate from the Green Lanes API token.

amberstarlight committed:
HMRC-1438: Update FAQ Feedback model to use new backend paths

amberstarlight committed:
HMRC-1438: Update FAQ Feedback

With the backend routes changing, the frontend routes need to change
also. They were not using standard RESTful routes, so this commit
changes them to live under `faq/feedback` and use GET and POST requests.

amberstarlight committed:
HMRC-1438: Decouple FAQ Feedback model from Green Lanes

willfish commented on app/controllers/green_lanes/faq_feedback_controller.rb:
```diff
@@ -0,0 +1,27 @@
+module GreenLanes
```
We actually don't need this namespace anymore

willfish commented on spec/factories/green_lanes/faq_feedback_factory.rb:
```diff
@@ -1,5 +1,5 @@
 FactoryBot.define do
-  factory :green_lanes_faq_feedback, class: 'GreenLanes::FaqFeedback' do
+  factory :green_lanes_faq_feedback, class: 'FaqFeedback' do
```
```suggestion
  factory :faq_feedback, class: 'FaqFeedback' do
```

willfish replied:
We actually don't want this file to be nested under the green_lanes scope anymore

willfish commented on spec/models/faq_feedback_spec.rb:
```diff
@@ -1,4 +1,4 @@
-RSpec.describe GreenLanes::FaqFeedback, type: :model do
+RSpec.describe FaqFeedback, type: :model do
```
We'll want to rename this file, too

willfish commented on config/routes.rb:
```diff
@@ -148,10 +148,7 @@
-    get 'get_feedback', to: 'faq#get_feedback'
-
-    post 'send_feedback', to: 'faq#send_feedback', as: :send_feedback
+    resources :faq_feedback, only: %i[index create], path: 'faq/feedback'
```
Nice

amberstarlight committed:
HMRC-1438: Code review

amberstarlight committed:
HMRC-1438: Fix broken tests

amberstarlight committed:
HMRC-1438: Change routing scope

Moves `green_lanes/faq/feedback` to `/faq_feedback`, which mirrors the
backend routing for this implementation.

willfish commented on spec/factories/faq_feedback_factory.rb:
```diff
@@ -1,5 +1,5 @@
 FactoryBot.define do
-  factory :green_lanes_faq_feedback, class: 'GreenLanes::FaqFeedback' do
+  factory :faq_feedback, class: 'FaqFeedback' do
```
We actually don't need the class anymore
```suggestion
  factory :faq_feedback do
```

amberstarlight committed:
HMRC-1438: Code review

willfish commented on app/controllers/green_lanes/faq_controller.rb:
```diff
@@ -1,36 +1,9 @@
-    # application contoller requires search params to be set
-    # override this for faq feedback ajax requests
-    skip_before_action :set_search, only: [:send_feedback]
+    skip_before_action :set_search, only: [:index]
```
So I think we can completely remove this controller

amberstarlight committed:
HMRC-1438: Remove skip_before_action
